pkg/kubevirt/machine.go

```diff
@@ -90,6 +94,23 @@ func NewMachine(ctx *context.MachineContext, client client.Client, namespace str
 		machine.vmInstance = vm
 	}
 
+	if vm != nil && vm.Spec.Template != nil {
+		for _, vol := range vm.Spec.Template.Spec.Volumes {
```

Comment:
sorry i didn't catch this earlier.

Technically we need to be looking for the presence of `vm.Spec.DataVolumeTemplates` here instead of `vm.Spec.Template.Spec.Volumes`. 

This is because the `Volumes` could be attaching the DataVolume as a PVC instead of a DataVolume. These are interchangable.

The source of truth in our case for whether a VM has a data import flow associated with it is to look at the DataVolumeTemplate section.